### PR TITLE
util-core: improve test coverage for Cancellable

### PR DESCRIPTION
**Pull request by muachilin**
Problem

The test coverage of Cancellable still needs to be improved. Particularly the object of Cancellable has not been tested.

Solution

Add more tests for Cancellable.

Result

Improve the test coverage for Cancellable.

### CODE REVIEW

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/10c0bc9811126fee707a0daa33d5b3b532afb099&el=desc) will **increase** coverage by `0.02%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #268      +/-   ##
===========================================
+ Coverage    45.97%   45.99%   +0.02%     
===========================================
  Files          228      228              
  Lines        14447    14447              
  Branches       917      917              
===========================================
+ Hits          6642     6645       +3     
+ Misses        7805     7802       -3     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `92.16% <0.00%> (ø)` | :arrow_up: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `78.31% <0.00%> (ø)` | :arrow_up: |
| [...ore/src/main/scala/com/twitter/util/Duration.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRHVyYXRpb24uc2NhbGE=) | `85.54% <0.00%> (+0.57%)` | :arrow_up: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `50.00% <0.00%> (+25.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [10c0bc9...4e7dd38](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/10c0bc9811126fee707a0daa33d5b3b532afb099&el=desc) will **increase** coverage by `0.02%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #268      +/-   ##
===========================================
+ Coverage    45.97%   46.00%   +0.02%     
===========================================
  Files          228      228              
  Lines        14447    14447              
  Branches       917      917              
===========================================
+ Hits          6642     6646       +4     
+ Misses        7805     7801       -4     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `92.16% <0.00%> (ø)` | :arrow_up: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `78.71% <0.00%> (+0.40%)` | :arrow_up: |
| [...ore/src/main/scala/com/twitter/util/Duration.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRHVyYXRpb24uc2NhbGE=) | `85.54% <0.00%> (+0.57%)` | :arrow_up: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `50.00% <0.00%> (+25.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [10c0bc9...4e7dd38](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/10c0bc9811126fee707a0daa33d5b3b532afb099&el=desc) will **increase** coverage by `0.00%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@           Coverage Diff            @@
##           develop     #268   +/-   ##
========================================
  Coverage    45.97%   45.98%           
========================================
  Files          228      228           
  Lines        14447    14447           
  Branches       917      917           
========================================
+ Hits          6642     6643    +1     
+ Misses        7805     7804    -1     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...in/scala/com/twitter/logging/QueueingHandler.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1sb2dnaW5nL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2xvZ2dpbmcvUXVldWVpbmdIYW5kbGVyLnNjYWxh) | `93.93% <0.00%> (-6.07%)` | :arrow_down: |
| [...l-core/src/main/scala/com/twitter/util/Event.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRXZlbnQuc2NhbGE=) | `92.05% <0.00%> (-0.67%)` | :arrow_down: |
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `91.95% <0.00%> (-0.21%)` | :arrow_down: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `78.31% <0.00%> (ø)` | :arrow_up: |
| [...ore/src/main/scala/com/twitter/util/Duration.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRHVyYXRpb24uc2NhbGE=) | `85.54% <0.00%> (+0.57%)` | :arrow_up: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `75.00% <0.00%> (+50.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [10c0bc9...f810905](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/10c0bc9811126fee707a0daa33d5b3b532afb099&el=desc) will **increase** coverage by `0.04%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #268      +/-   ##
===========================================
+ Coverage    45.97%   46.01%   +0.04%     
===========================================
  Files          228      228              
  Lines        14447    14447              
  Branches       917      917              
===========================================
+ Hits          6642     6648       +6     
+ Misses        7805     7799       -6     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...in/scala/com/twitter/logging/QueueingHandler.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1sb2dnaW5nL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2xvZ2dpbmcvUXVldWVpbmdIYW5kbGVyLnNjYWxh) | `93.93% <0.00%> (-6.07%)` | :arrow_down: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `78.71% <0.00%> (+0.40%)` | :arrow_up: |
| [...ore/src/main/scala/com/twitter/util/Duration.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRHVyYXRpb24uc2NhbGE=) | `85.54% <0.00%> (+0.57%)` | :arrow_up: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `100.00% <0.00%> (+75.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [10c0bc9...471fbe4](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/10c0bc9811126fee707a0daa33d5b3b532afb099&el=desc) will **increase** coverage by `0.03%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #268      +/-   ##
===========================================
+ Coverage    45.97%   46.00%   +0.03%     
===========================================
  Files          228      228              
  Lines        14447    14447              
  Branches       917      917              
===========================================
+ Hits          6642     6647       +5     
+ Misses        7805     7800       -5     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...in/scala/com/twitter/logging/QueueingHandler.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1sb2dnaW5nL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2xvZ2dpbmcvUXVldWVpbmdIYW5kbGVyLnNjYWxh) | `93.93% <0.00%> (-6.07%)` | :arrow_down: |
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `91.95% <0.00%> (-0.21%)` | :arrow_down: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `79.11% <0.00%> (+0.80%)` | :arrow_up: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `100.00% <0.00%> (+75.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [10c0bc9...1ad3a15](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 1]
```diff
@@ -3,6 +3,39 @@ package com.twitter.util
 import org.scalatest.WordSpec
 
 class CancellableTest extends WordSpec {
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: Hi @muachilin, thanks for these changes! Would you mind changing this over to FunSpec? This is noted in our [CONTRIBUTING.md](https://github.com/twitter/util/blob/develop/CONTRIBUTING.md#testing). I apologize that this file wasn't moved over to FunSpec, but we would greatly appreciate if would make that change.

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 2]
```diff
@@ -3,6 +3,39 @@ package com.twitter.util
...
+  "Cancellable" should {
+    "test variable isCancelled" in {
+      assert(!Cancellable.nil.isCancelled)
+    }
+    "test method cancel" in {
+      var cancelMethodCalled = false
+      class CancellableMethodCancelTest extends Cancellable{
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: nit: space `Cancellable {`

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 3]
```diff
@@ -3,6 +3,39 @@ package com.twitter.util
...
+        def linkTo(other: Cancellable): Unit = {}
+      }
+      val s = new CancellableMethodCancelTest()
+      assert(!cancelMethodCalled)
+      s.cancel()
+      assert(cancelMethodCalled)
+    }
+    "test method linkTo" in {
+      var linkToMethodCalled = false
+      class CancellableMethodLinkToTest extends Cancellable{
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: nit: space `Cancellable {`

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 4]
```diff
@@ -3,6 +3,39 @@ package com.twitter.util
 import org.scalatest.WordSpec
 
 class CancellableTest extends WordSpec {
+  "Cancellable" should {
+    "test variable isCancelled" in {
+      assert(!Cancellable.nil.isCancelled)
+    }
+    "test method cancel" in {
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: nit: please ensure there is line spacing between the tests (see [FlaggableTest](https://github.com/twitter/util/blob/develop/util-app/src/test/scala/com/twitter/app/FlaggableTest.scala) as an example)

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 5]
```diff
@@ -13,5 +46,30 @@ class CancellableTest extends WordSpec {
       s.cancel()
       assert(count == 1)
     }
+    "has not cancelled" in {
+      var count = 0
+      def increment: Unit = count += 1
+      val s = new CancellableSink(increment)
+      assert(!s.isCancelled)
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: should we also `assert(count == 0)`?

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 6]
```diff
@@ -13,5 +46,30 @@ class CancellableTest extends WordSpec {
...
+    }
+    "confirm cancelled" in {
+      var count = 0
+      def increment: Unit = count += 1
+      val s = new CancellableSink(increment)
+      s.cancel()
+      assert(s.isCancelled)
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: should we also `assert(count ==1)`?

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala [hunk 7]
```diff
@@ -3,6 +3,39 @@ package com.twitter.util
 import org.scalatest.WordSpec
 
 class CancellableTest extends WordSpec {
+  "Cancellable" should {
+    "test variable isCancelled" in {
+      assert(!Cancellable.nil.isCancelled)
+    }
```
Comment (enbnt) [inline util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: This is a test on the `Cancellable` companion object's `nil` property and not the `Cancellable` interface, correct? Should we give it a clearer test title (i.e. `"test Cancellable.nil is always cancelled"`)?

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala (hunk 1 above)
Comment (muachilin) [reply util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: Okay!! :)

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala (hunk 5 above)
Comment (muachilin) [reply util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: No problem!

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala (hunk 6 above)
Comment (muachilin) [reply util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: No problem!

File: util-core/src/test/scala/com/twitter/util/CancellableTest.scala (hunk 7 above)
Comment (muachilin) [reply util-core/src/test/scala/com/twitter/util/CancellableTest.scala]: Okay!

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/10c0bc9811126fee707a0daa33d5b3b532afb099&el=desc) will **increase** coverage by `0.05%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #268      +/-   ##
===========================================
+ Coverage    45.97%   46.03%   +0.05%     
===========================================
  Files          228      228              
  Lines        14447    14447              
  Branches       917      917              
===========================================
+ Hits          6642     6650       +8     
+ Misses        7805     7797       -8     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `91.95% <0.00%> (-0.21%)` | :arrow_down: |
| [...ore/src/main/scala/com/twitter/util/Duration.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRHVyYXRpb24uc2NhbGE=) | `85.54% <0.00%> (+0.57%)` | :arrow_up: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `79.11% <0.00%> (+0.80%)` | :arrow_up: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `100.00% <0.00%> (+75.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [10c0bc9...55e6a21](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by muachilin** [comment]
Is there anything that I need to modify in the tests for Cancellable? Thank you!

**Comment by ryanoneill** [comment]
Hi @muachilin, would you mind rebasing off of the latest master? If so, then I can get this merged in internally. Thanks.

**Comment by muachilin** [comment]
Okay!

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/2b0b3dc040d3ca26c1bfdedf977e78dddd9fccb7&el=desc) will **decrease** coverage by `0.01%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop     #268      +/-   ##
===========================================
- Coverage    46.04%   46.03%   -0.02%     
===========================================
  Files          228      228              
  Lines        14443    14443              
  Branches       922      922              
===========================================
- Hits          6651     6649       -2     
- Misses        7792     7794       +2     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...in/scala/com/twitter/logging/QueueingHandler.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1sb2dnaW5nL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2xvZ2dpbmcvUXVldWVpbmdIYW5kbGVyLnNjYWxh) | `93.93% <0.00%> (-6.07%)` | :arrow_down: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `77.95% <0.00%> (-0.82%)` | :arrow_down: |
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `91.75% <0.00%> (-0.62%)` | :arrow_down: |
| [...ore/src/main/scala/com/twitter/util/Duration.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvRHVyYXRpb24uc2NhbGE=) | `84.97% <0.00%> (-0.58%)` | :arrow_down: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `100.00% <0.00%> (+75.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [2b0b3dc...8a6a176](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by caniszczyk** [comment]
# [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=h1) Report
> Merging [#268](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=desc) into [develop](https://codecov.io/gh/twitter/util/commit/2b0b3dc040d3ca26c1bfdedf977e78dddd9fccb7&el=desc) will **increase** coverage by `0.00%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree)

```diff
@@           Coverage Diff            @@
##           develop     #268   +/-   ##
========================================
  Coverage    46.04%   46.05%           
========================================
  Files          228      228           
  Lines        14443    14443           
  Branches       922      922           
========================================
+ Hits          6651     6652    +1     
+ Misses        7792     7791    -1     
```


| [Impacted Files](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...in/scala/com/twitter/logging/QueueingHandler.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1sb2dnaW5nL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2xvZ2dpbmcvUXVldWVpbmdIYW5kbGVyLnNjYWxh) | `93.93% <0.00%> (-6.07%)` | :arrow_down: |
| [util-core/src/main/scala/com/twitter/io/Buf.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL2lvL0J1Zi5zY2FsYQ==) | `91.95% <0.00%> (-0.42%)` | :arrow_down: |
| [...core/src/main/scala/com/twitter/util/Promise.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvUHJvbWlzZS5zY2FsYQ==) | `78.36% <0.00%> (-0.41%)` | :arrow_down: |
| [.../src/main/scala/com/twitter/util/Cancellable.scala](https://codecov.io/gh/twitter/util/pull/268/diff?src=pr&el=tree#diff-dXRpbC1jb3JlL3NyYy9tYWluL3NjYWxhL2NvbS90d2l0dGVyL3V0aWwvQ2FuY2VsbGFibGUuc2NhbGE=) | `100.00% <0.00%> (+75.00%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=footer). Last update [2b0b3dc...8a6a176](https://codecov.io/gh/twitter/util/pull/268?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by ryanoneill** [comment]
Sorry, I meant to say 'develop' branch, not 'master'. Also, it seems like there are still some old commits in the pull request beyond the 'CancellableTest' one which is the problem. The goal is to have the [patch url](https://patch-diff.githubusercontent.com/raw/twitter/util/pull/268.patch) be just your test change here, otherwise it won't apply cleanly internally. When that's the case we can merge it in and make sure you get proper attribution. Sorry for the hoops to jump through.

**Comment by muachilin** [comment]
I see. Can I create another branch for simply the "Cancellable Test"? Since my current branch has other commits (Improve test coverage in StorageUnit)... Sorry!! I'll close this PR and open another one instead.

**Pull request closed without merging**